cli/cmd/login.go

```diff
@@ -0,0 +1,167 @@
+package cmd
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"github.com/adrg/xdg"
+	"github.com/cenkalti/backoff/v4"
+	"github.com/pkg/browser"
+	"github.com/spf13/cobra"
+)
+
+const (
+	loginShort = "Login to CloudQuery Hub."
+	loginLong  = `Login to CloudQuery Hub.
+
+This is required to download plugins from CloudQuery Hub.
+
+Local plugins and different registries don't need login.
+`
+
+	accountsURL = "https://accounts.cloudquery.io"
+)
+
+func newCmdLogin() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:    "login",
+		Short:  loginShort,
+		Long:   loginLong,
+		Hidden: true,
+		Args:   cobra.MatchAll(cobra.ExactArgs(0), cobra.OnlyValidArgs),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			// Set up a channel to listen for OS signals for graceful shutdown.
+			ctx, cancel := context.WithCancel(cmd.Context())
+
+			sigChan := make(chan os.Signal, 1)
+			signal.Notify(sigChan, syscall.SIGTERM)
+
+			go func() {
+				<-sigChan
+				fmt.Println("Received SIGTERM!")
+				cancel()
+			}()
+
+			return runLogin(ctx)
+		},
+	}
+	return cmd
+}
+
+func waitForServer(ctx context.Context, url string) error {
+	ctx, cancel := context.WithTimeout(ctx, 3*time.Second)
+	defer cancel()
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, url, http.NoBody)
+	if err != nil {
+		return err
+	}
+
+	return backoff.Retry(func() error {
+		resp, err := http.DefaultClient.Do(req)
+		if err != nil {
+			return err
+		}
+		defer resp.Body.Close()
+		_, _ = io.Copy(io.Discard, resp.Body)
+		if resp.StatusCode == http.StatusOK {
+			return nil
+		}
+		return fmt.Errorf("failed to connect to local server. error code: %d", resp.StatusCode)
+	}, backoff.WithContext(backoff.NewConstantBackOff(100*time.Millisecond), ctx))
+}
+
+func runLogin(ctx context.Context) (err error) {
+	tokenFilePath, err := xdg.CacheFile("cloudquery/token")
```

Comment:
I don't think `CacheFile` is the correct method to use, see https://wiki.archlinux.org/title/XDG_Base_Directory

![image](https://github.com/cloudquery/cloudquery/assets/26760571/30f26439-ff17-426b-93bb-7ff747580617)

Reply:
`.cq` would make more sense for cache directory

Reply:
yeah, it's a typo (I wanted `xdg.ConfigFile`)

Reply:
Actually we might want to use `DataHome` https://github.com/mislav/hub/issues/1912

Reply:
Done in 6a1a2ab895f36ccb57452b7c582a7835cfae16a7